consensus/src/epoch_manager.rs

```diff
@@ -419,6 +440,17 @@ impl EpochManager {
 
         let safety_rules_container = Arc::new(Mutex::new(safety_rules));
 
+        let (consensus_to_quorum_store_sender, consensus_to_quorum_store_receiver) =
+            mpsc::channel(self.config.intra_consensus_channel_buffer_size);
+        self.spawn_quorum_store(consensus_to_quorum_store_receiver);
+        let payload_manager = QuorumStoreClient::new(
+            consensus_to_quorum_store_sender.clone(),
+            self.config.quorum_store_poll_count,
+            self.config.quorum_store_pull_timeout_ms,
+        );
+        self.commit_notifier
+            .new_epoch(consensus_to_quorum_store_sender);
```

Comment:
I think this implicitly drop the previous sender and terminates the previous quorum store, it may be better to explicit drop them inside the shutdown_current_processor function